pyquil/parameters.py

```diff
@@ -158,6 +158,43 @@ def __eq__(self, other):
         return isinstance(other, Parameter) and other.name == self.name
 
 
+class Segment(Parameter):
```

Comment:
I don't think this should subclass Parameter, it's very different than eg `%theta`

Reply:
Would you find subclassing `Expression` more appropriate?